api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2499,8 +2499,42 @@ public void copyObject(CopyObjectArgs args)
             null,
             null,
             0)) {
-      // For now ignore the copyObjectResult, just read and parse it.
-      Xml.unmarshal(CopyObjectResult.class, response.body().charStream());
+
+      String bodyContent = new String(response.body().bytes(), StandardCharsets.UTF_8);
+      bodyContent = bodyContent.trim();
+      if (!bodyContent.isEmpty()) {
+        try {
+          if (Xml.validate(ErrorResponse.class, bodyContent)) {
+            ErrorResponse errorResponse = Xml.unmarshal(ErrorResponse.class, bodyContent);
+            throw new ErrorResponseException(errorResponse, response);
+          }
+        } catch (XmlParserException e) {
+          // As it is not <Error> message, fall-back to parse CopyObjectResult XML.
```

Comment:
This is `CompleteMultipartUpload` behaviour i.e. `Error` is returned with `200 OK`. Here, you need to do XML parsing for `CopyObjectResult`